pkg.nix

```diff
@@ -99,15 +98,20 @@ in stdenv.mkDerivation (finalAttrs: {
 
   configureFlags = [
     "--with-system-luarocks"
-    # TODO: Explain this flag
+    # The automake check target uses `pdfinfo` to confirm the output of a test
+    # run. This flake build eschews that test because it is run from the source
+    # directory but the binary is already built with system paths, so it can't
+    # be checked under Nix until after install. By specifying some binary here
+    # we skip the configure time test for `pdfinfo`, by using `false` we make
+    # sure that check fails if run (which we don't).
```

Comment:
But in Nixpkgs we do run this check? If we do, then perhaps that would be worth mentioning as another difference between the two expressions.

Reply:
The derivation does run an equivalent check, but not straight out of the Makefile (which isn't available after installation when we actually want to test it). Instead we have a copy of the shell code from the target that we run in `passthru.tests.test` and run after everything is in the expected location. In the shell code there we have hard coded `pdfinfo` instead of the upstream `$(PDFINFO)` set by autoconf.

Reply:
I added that nugget of info to the comment and also a few other cleanups. It seems to work now (with the exception of Lua 5.4 where `lua-zlib` is marked as broken. The LuaJIT, Lua5_2, and Lua5_3 options all run correctly.